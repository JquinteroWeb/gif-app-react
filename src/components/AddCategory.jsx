import { useState } from 'react';


export const AddCategory = () => {
    const [inputValue, setInputValue] = useState('Dragon Ball');

    const onInput = ({ target }) => {
        setInputValue(target.value);
    }
    const onSubmit = (event) =>{
        event.preventDefault();       
    }
    return (
        <form onSubmit={onSubmit}>
            <div>
                <input
                    type="text"
                    placeholder="Search Gifs"
                    value={inputValue}
                    onChange={onInput}
                />
            </div>
        </form>
    )
}

