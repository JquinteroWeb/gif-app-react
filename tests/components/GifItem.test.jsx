import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItiem />', () => {

    const title = 'Dragon ball';
    const url = 'https://github.com/';
    
    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el url y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        //screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);        
        const {src , alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar titulo', () => {
        render(<GifItem title={title} url={url} />);       
        expect(screen.getByText(title)).toBeTruthy();   
    });
});