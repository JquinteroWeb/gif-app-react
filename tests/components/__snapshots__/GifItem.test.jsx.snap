// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItiem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="https://github.com/"
    />
    <p>
      Dragon ball
    </p>
  </div>
</div>
`;
